src/main/java/seedu/club/storage/JsonClubBookStorage.java

```diff
@@ -18,147 +18,143 @@
 
-        Optional<JsonSerializableEvent> jsonEvents = JsonUtil.readJsonFile(
-                eventFilePath, JsonSerializableEvent.class);
+        Optional<JsonSerializableClubBook> jsonEvents = JsonUtil.readJsonFile(
+                clubBookFilePath, JsonSerializableClubBook.class);
         if (!jsonEvents.isPresent()) {
             return Optional.empty();
         }
 
         try {
             return Optional.of(jsonEvents.get().toModelType());
         } catch (IllegalValueException ive) {
-            logger.info("Illegal values found in " + eventFilePath + ": " + ive.getMessage());
+            logger.info("Illegal values found in " + clubBookFilePath + ": " + ive.getMessage());
             throw new DataLoadingException(ive);
         }
     }
 
     @Override
     public void saveEvents(ReadOnlyClubBook clubBook) throws IOException {
-        saveEvents(clubBook, eventFilePath);
+        saveEvents(clubBook, clubBookFilePath);
     }
 
     /**
      * Similar to {@link #saveEvents(ReadOnlyClubBook)}.
      *
-     * @param eventFilePath location where event data is to be stored. Cannot be null.
+     * @param clubBookFilePath location where event data is to be stored. Cannot be null.
      */
-    public void saveEvents(ReadOnlyClubBook clubBook, Path eventFilePath) throws IOException {
+    public void saveEvents(ReadOnlyClubBook clubBook, Path clubBookFilePath) throws IOException {
         requireNonNull(clubBook);
-        requireNonNull(eventFilePath);
+        requireNonNull(clubBookFilePath);
 
-        FileUtil.createIfMissing(eventFilePath);
-        JsonUtil.saveJsonFile(new JsonSerializableEvent(clubBook), eventFilePath);
+        FileUtil.createIfMissing(clubBookFilePath);
+        JsonUtil.saveJsonFile(new JsonSerializableClubBook(clubBook), clubBookFilePath);
     }
 
     // ================ Club Book methods ==============================
 
     /**
      * Reads data from both member and event files and store them in a new ClubBook
      *
-     * @param memberFilePath location where member data is stored
-     * @param eventFilePath location where event data is stored
      * @return {@code Optional<ReadOnlyClubBook>} containing both member and event data
      * @throws DataLoadingException if loading from storage fails
      */
-    public Optional<ReadOnlyClubBook> readFullClubBook(Path memberFilePath,
-                                                       Path eventFilePath) throws DataLoadingException {
-        requireNonNull(memberFilePath);
-        requireNonNull(eventFilePath);
+    public Optional<ReadOnlyClubBook> readClubBook() throws DataLoadingException {
+        requireNonNull(clubBookFilePath);
+        requireNonNull(clubBookFilePath);
 
         Optional<ReadOnlyClubBook> members = readMembers();
         Optional<ReadOnlyClubBook> events = readEvents();
 
+        if (members.isEmpty() || events.isEmpty()) {
+            return Optional.empty();
+        }
```

Comment:
wait correct me if im wrong but if members has data but events has no data (vice versa), we display nothing?

Shouldn't it be `&&`, if either one has data or both has data we shld still return `Optional.of(clubBook)`
```suggestion
        if (members.isEmpty() && events.isEmpty()) {
            return Optional.empty();
        }
```